### PR TITLE
Laravel 10.x Compatibility

### PR DESCRIPTION
**Pull request by laravel-shift**
This is an automated pull request from [Shift](https://laravelshift.com) to update your package code and dependencies to be compatible with Laravel 9.x.

**Before merging**, you need to:

- Checkout the `l10-compatibility` branch
- Review **all** comments for additional changes
- Thoroughly test your package

If you do find an issue, please report it by commenting on this PR to help improve future automation.

### CODE REVIEW

**Comment by laravel-shift** [comment]
:warning: Shift detected a dependency for the `laravel/framework`. Unless you have a special requirement for depending on the entire Laravel framework, it's recommended to add dependencies for the specific `illuminate` packages instead.

**Comment by laravel-shift** [comment]
:alembic: **Using this package?** If you would like to help test these changes or believe them to be compatible, you may update your project to reference this branch.

To do so, temporarily add Shift's fork to the `repositories` property of your `composer.json`:

```json
{
    "repositories": [
        {
            "type": "vcs",
            "url": "https://github.com/laravel-shift/larinfo.git"
        }
    ]
}
```

Then update your dependency constraint to reference this branch:

```json
{
    "require": {
        "matriphe/larinfo": "dev-l10-compatibility",
    }
}
```

Finally, run: `composer update`

**Comment by dietercoopman** [comment]
Hey, @matriphe , we depend on your package in our application and want to upgrade to L10.  Can we help you to get this pull request approved.  It's been automatically shifted by Laravel shift and it looks ok.

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -56,8 +56,8 @@ jobs:
       fail-fast: false
       matrix:
         operating-system: [ ubuntu-latest ]
-        php-versions: [ '7.4' ]
-        laravel-versions: [ '5.7.*', '^6.0', '^7.0', '^8.0' ]
+        php-versions: [8.1]
+        laravel-versions: [10.*]
```
Comment (matriphe) [inline .github/workflows/master.yml:60]: please don't update this. it's for testing the older version.

File: .github/workflows/master.yml [hunk 2]
```diff
@@ -78,8 +78,8 @@ jobs:
       fail-fast: false
       matrix:
         operating-system: [ ubuntu-latest ]
-        php-versions: [ '7.3', '7.4' ]
-        laravel-versions: [ '5.6.*' ]
+        php-versions: [8.1]
+        laravel-versions: [10.*]
```
Comment (matriphe) [inline .github/workflows/master.yml:82]: please don't update this as well. it's for testing the older version.

File: composer.json [hunk 3]
```diff
@@ -2,23 +2,23 @@
...
+    "keywords": [
         "laravel",
         "larinfo",
         "ipinfo",
         "linfo",
         "sysinfo"
     ],
     "require": {
-        "php": "^8.0",
+        "php": "^8.1",
```
Comment (matriphe) [inline composer.json:13]: maybe?
```suggestion
        "php": "8.*",
```

File: composer.json [hunk 4]
```diff
@@ -2,23 +2,23 @@
...
         "ext-pdo": "*",
         "davidepastore/ipinfo": "^0.6",
-        "laravel/framework": "^9.0",
+        "laravel/framework": "^10.0",
```
Comment (matriphe) [inline composer.json:16]: maybe
```suggestion
        "laravel/framework": "^9.0|^10.0",
```

File: composer.json [hunk 5]
```diff
@@ -2,23 +2,23 @@
...
         "linfo/linfo": "^4.0"
     },
     "require-dev": {
-        "mockery/mockery": "^1.0",
-        "orchestra/testbench": "^7.0"
+        "mockery/mockery": "^1.4.4",
+        "orchestra/testbench": "^8.0"
```
Comment (matriphe) [inline composer.json:21]: maybe
```suggestion
        "mockery/mockery": "1.*",
        "orchestra/testbench": "^7.0|^8.0"
```

File: .github/workflows/master.yml [hunk 6]
```diff
@@ -22,8 +22,8 @@ jobs:
       fail-fast: false
       matrix:
         operating-system: [ ubuntu-latest ]
-        php-versions: [ '8.0', '8.1' ]
-        laravel-versions: [ '^9.0' ]
+        php-versions: [8.1]
+        laravel-versions: [10.0]
```
[the remark this answers is not in the document]
Comment (matriphe) [reply .github/workflows/master.yml:26]: I just read the Laravel 10 release, and it requires PHP 8.1. I'd suggest creating new jobs for testing Laravel 10.

**Comment by matriphe** [comment]
Closing this in favor of [version 4.1.0](https://github.com/matriphe/larinfo/releases/tag/4.1.0) which supports Laravel 10.

Thank you for your contribution!

**Pull request closed without merging**